src/MusicStore/Components/ISystemClock.cs

```diff
@@ -0,0 +1,16 @@
+﻿using System;
+
+namespace MusicStore.Components
+{
+    /// <summary>
+    /// Abstracts the system clock to facilitate testing.
+    /// </summary>
+    public interface ISystemClock
```

Comment:
@Eilon 
I added this interface, and the method UtcNow returns DateTime, but...I'm thinking this name is not good here or we should return DateTimeOffset from this interface. This interface should be used not only for Album, but it also could be used for the other models where DateTimeOffset is more apporiate. 
I'd like to change its method to return DateTimeOffset instead of DateTime..What do you think??